packages/node-core/src/configure/migration-service/migration.ts

```diff
@@ -145,47 +280,231 @@ export class Migration {
+        return;
+      }
+
+      switch (relation.type) {
+        case 'belongsTo': {
+          model.belongsTo(relatedModel, {foreignKey: relation.foreignKey});
+          addRelationToMap(relation, this.foreignKeyMap, model, relatedModel);
+          const rel = model.belongsTo(relatedModel, {foreignKey: relation.foreignKey});
+          const fkConstraint = getFkConstraint(rel.source.tableName, rel.foreignKey);
+          if (this.dbType !== SUPPORT_DB.cockRoach) {
+            this.extraQueries.push(constraintDeferrableQuery(model.getTableName().toString(), fkConstraint));
+          }
+          break;
+        }
+        case 'hasOne': {
+          const rel = model.hasOne(relatedModel, {
+            foreignKey: relation.foreignKey,
+          });
+          const fkConstraint = getFkConstraint(rel.target.tableName, rel.foreignKey);
+          const tags = smartTags({
+            singleForeignFieldName: relation.fieldName,
+          });
+          this.extraQueries.push(
+            commentConstraintQuery(`"${this.schemaName}"."${rel.target.tableName}"`, fkConstraint, tags),
+            createUniqueIndexQuery(this.schemaName, relatedModel.tableName, relation.foreignKey)
+          );
+          break;
+        }
+        case 'hasMany': {
+          const rel = model.hasMany(relatedModel, {
+            foreignKey: relation.foreignKey,
+          });
+          const fkConstraint = getFkConstraint(rel.target.tableName, rel.foreignKey);
+          const tags = smartTags({
+            foreignFieldName: relation.fieldName,
+          });
+          this.extraQueries.push(
+            commentConstraintQuery(`"${this.schemaName}"."${rel.target.tableName}"`, fkConstraint, tags)
+          );
+          break;
+        }
+        default:
+          throw new Error('Relation type is not supported');
+      }
+      const relationQuery = generateForeignKeyStatement(model.getAttributes()[relation.foreignKey], model.tableName);
+      if (relationQuery) {
+        this.mainQueries.push(relationQuery);
+      }
+    }
+
+    this.addModelToSequelizeCache(model);
+  }
+
+  private addRelationComments(): void {
+    this.foreignKeyMap.forEach((keys, tableName) => {
+      const comment = Array.from(keys.values())
+        .map((tags) => smartTags(tags, '|'))
+        .join('\n');
+      const query = commentTableQuery(`"${this.schemaName}"."${tableName}"`, comment);
+      this.extraQueries.push(query);
+    });
+  }
+
+  dropRelation(relation: GraphQLRelationsType): void {
+    if (relation.type !== 'belongsTo') {
+      return;
+    }
+    const tableName = modelToTableName(relation.from);
+    const fkConstraint = getFkConstraint(tableName, relation.foreignKey);
+
+    this.mainQueries.unshift(dropRelationStatement(this.schemaName, tableName, fkConstraint));
+
+    // TODO remove from sequelize model
+    // Should update sequelize model cache
+  }
+
+  async createEnum(e: GraphQLEnumsType): Promise<void> {
```

Comment:
I think this function should be reworked to be non-async so that it matches the other functions. 

We can move the async stuff to the create function.